tests/fe/fe_nedelec_face_edge_orientation.h

```diff
@@ -0,0 +1,451 @@
+// opposite directions. That is, the least significant bit (LSB) of the
+// combined_face_orientation parameter is interpreted as the
+// line_orientation parameter which is used in deal.II on two-dimensional
+// meshes.
+//
+// Test meshes in two and three dimensions are generated by functions
+// GridGenerator::non_standard_orientation_mesh().
+// The three-dimensional mesh is used as it is. Only the leftmost shared face
+// is used in two dimensions. That is, shape functions are sampled only on
+// the leftmost and the middle cells of the two-dimensional mesh.
+
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_nedelec.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_orientation.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/fe_field_function.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <cmath>
+#include <string>
+
+#include "../tests.h"
+
+template <int dim>
+void
+create_triangulation(Triangulation<dim> &triangulation,
+                     const unsigned char combined_face_orientation);
+
+template <int dim>
+void
+create_sampling_points(std::vector<Point<dim>> &sampling_points_left,
+                       std::vector<Point<dim>> &sampling_points_right,
+                       unsigned int             N);
+
+template <int dim>
+void
+save_sampling_points(std::vector<Point<dim>> &sampling_points_left,
+                     std::vector<Point<dim>> &sampling_points_right);
+
+template <int dim>
+void
+print_shared_face(Triangulation<dim> &triangulation);
+
+template <>
+void
+create_triangulation<2>(Triangulation<2>   &triangulation,
+                        const unsigned char combined_face_orientation)
+
+{
+  // Only the leftmost shared face is considered. Even values of
+  // combined_face_orientation produce misaligned faces (the middle square is
+  // rotated by 2*[pi/2]). Odd values of combined_face_orientation produce
+  // aligned faces (the middle square is not rotated).
+  const unsigned int n_rotate_middle_square =
+    (combined_face_orientation % 2 == 0) ? 2 : 0;
+
+  GridGenerator::non_standard_orientation_mesh(triangulation,
+                                               n_rotate_middle_square);
+}
+
+template <>
+void
+create_triangulation<3>(Triangulation<3>   &triangulation,
+                        const unsigned char combined_face_orientation)
+
+{
+  bool face_orientation;
+  bool face_rotation;
+  bool face_flip;
+
+  std::tie(face_orientation, face_rotation, face_flip) =
+    dealii::internal::split_face_orientation(combined_face_orientation);
+
+  GridGenerator::non_standard_orientation_mesh(
+    triangulation, face_orientation, face_flip, face_rotation, false);
+}
+
+template <>
+void
+create_sampling_points<2>(std::vector<Point<2>> &sampling_points_left,
```

Comment:
what is your goal for setting these sampling points? Would it be possible to use a deal.II Quadrature like ``QIterated(N, QMidpoint())`` or something similar?

Reply:
The sampling points are chosen in two neighboring cells such that their spatial positions and indices reflect correctly in the shared face. The figure below illustrates the idea. 

![shared_face](https://github.com/user-attachments/assets/5ca63545-e373-4102-88f4-ec979d225761)

I can use QIterated to create the points at the left cell. But then I need to duplicate the quadrature points and loop through the duplicated points to shift their positions and reflect them in the shared face. My thinking was: as soon as I need to loop anyway, I may as well create the sampling points from scratch.

Do you know an elegant way to reflect the quadrature points in a shared face?